src/memory.c

```diff
@@ -136,6 +136,16 @@ alloc_array( size_t item_count, size_t item_size ) {
   return alloc_mem(item_count * item_size);
 }
 
+void *
+realloc_array( const void *mem, size_t item_count, size_t item_size ) {
+  #include <stdio.h>
```

Comment:
this include may be left over from testing? At any rate, if it is still needed it shouldn't be inside of this function.

Reply:
I'm sorry about this: I've really forgot to remove this header.

I've just refactored the 3 commits of this PR. Everything should be ok now.